### PR TITLE
feat: Eslint 9 + flat config support

### PR DESCRIPTION
**Pull request by timofei-iatsenko**
Fixes: https://github.com/lingui/eslint-plugin/issues/46 https://github.com/lingui/eslint-plugin/issues/31 https://github.com/lingui/eslint-plugin/issues/29 https://github.com/lingui/eslint-plugin/issues/9 https://github.com/lingui/eslint-plugin/issues/5

- Support eslint v9 (as well as v8)
- Support Flat Configs
- Added a recomended config
- Updated all dependendencies
- Split rule docs to separate files and referenced them from the rules.

### CODE REVIEW

File: src/rules/no-unlocalized-strings.ts [hunk 1]
```diff
@@ -408,16 +402,17 @@ const rule: RuleModule<string, Option[]> = {
         //
         // TYPESCRIPT
         //
+        if (option?.useTsTypes) {
+          const services = ESLintUtils.getParserServices(context)
```
Comment (timofei-iatsenko) [inline src/rules/no-unlocalized-strings.ts:406]: https://typescript-eslint.io/developers/custom-rules#conditional-type-information

**Comment by andrii-bodnar** [comment]
Awesome, thank you!

I feel like it might be a good time to do a major v1 release and remove the `beta` label.

This could also be a great addition to the Lingui v5 release notes. What do you think?

**Comment by timofei-iatsenko** [comment]
I think we can do v1 release after https://github.com/lingui/eslint-plugin/issues/50

Regarding this PR, i'm testing it right now with our project and found couple issues with build setup, i will turn this one into draft and mark as ready when finish.

**Comment by timofei-iatsenko** [comment]
@andrii-bodnar ready. Tested on my project.

**Comment by timofei-iatsenko** [comment]
@andrii-bodnar let's merge and publish these two PRs as quickly as possible. I'm working on an integration for my new project and these things is blocking me.

**Review by andrii-bodnar — APPROVED** [review]

**Pull request merged**